gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinTaskRunner.java

```diff
@@ -196,6 +196,11 @@ public GobblinTaskRunner(String applicationName,
     this.services.addAll(suite.getServices());
 
     this.services.addAll(getServices());
+
+    if (ConfigUtils.getBoolean(this.config, GobblinClusterConfigurationKeys.CONTAINER_HEALTH_METRICS_SERVICE_ENABLED, false)) {
```

Comment:
Should we package this service as part of `TaskRunnerSuiteBase` instead of having a `this.services.addAll(getServices)` plus another service which is used for metric-reporting outside?

Reply:
The TaskRunnerSuiteBase is an abstract class with two implementations, process model and thread model. Wanted to leave getServices() in TaskRunnerSuiteBase as an abstract method so as not to change the contract of the class.

Reply:
Sorry I should not mentioned `TaskRunnerSuiteBase`. What I meant to say is, does it make more sense to add this service inside `getServices` method in `GobblinTaskRunner` ?  

The comment of `getServices` is :
```Creates and returns a {@link List} of additional {@link Service}s that should be run in this {@link GobblinTaskRunner}. Sub-classes that need additional {@link Service}s to run, should override this method```